ml-proto/README.md

```diff
@@ -5,10 +5,13 @@ This repository implements a prototypical reference interpreter for WebAssembly.
 Currently, it can
 
 * *parse* a simple S-expression format,
+* *decode* the binary format (work in progress),
 * *validate* modules defined in it,
-* *execute* invocations to functions exported by a module.
+* *execute* invocations to functions exported by a module,
+* *encode* the binary format,
+* *prettyprint* the S-expression format (work in progress).
 
-The file format is a (very dumb) form of *script* that cannot just define a module, but also batch a sequence of invocations.
+The S-expression format is a (very dumb) form of *script* that cannot just define a module, but also batch a sequence of invocations, assertions, and conversions. As such it is a superset of the binary format, with the additional functionality purely intended as testing infrastructure.
```

Comment:
Good point. Added most of that to the section on scripts, and only a short remark and a forward reference here.